codersdk/agentsdk/logs.go

```diff
@@ -0,0 +1,219 @@
+package agentsdk
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"io"
+	"net/http"
+	"time"
+
+	"golang.org/x/xerrors"
+
+	"cdr.dev/slog"
+	"github.com/coder/coder/codersdk"
+	"github.com/coder/retry"
+)
+
+type startupLogsWriter struct {
```

Comment:
Review: I went a round of back-and-forth on making this concurrently safe. Ultimately I opted to revert the concurrency safety in favor of keeping the implementation as short/simple as possible ([`cdb64e2` (#8129)](https://github.com/coder/coder/pull/8129/commits/cdb64e203bca4569f0071304c4063ffc3713dc23)).